scripts/apollo-on-kubernetes/apollo-admin-server/Dockerfile

```diff
@@ -24,6 +24,6 @@ ENV DATASOURCES_PASSWORD="FillInCorrectPassword"
 
 EXPOSE 8090
 
-ENTRYPOINT ["/apollo-admin-server/entrypoint.sh"]
+#ENTRYPOINT ["/apollo-admin-server/entrypoint.sh"]
```

Comment:
the logic of entrypoint.sh is to fill in the database connection information, if this line is commentted, then the database connection information will be missed?

Reply:
数据库连接信息是通过configmap对象提供的，然后通过volumes挂载到容器中，实际上挂载的是文件application-github.properties

请参考文件apollo/scripts/apollo-on-kubernetes/kubernetes/apollo-env-dev/service-apollo-admin-server-dev.yaml

```yaml
#定义configmap
kind: ConfigMap
apiVersion: v1
metadata:
  namespace: sre
  name: configmap-apollo-admin-server-dev
data:
  application-github.properties: |
    spring.datasource.url = jdbc:mysql://service-mysql-for-apollo-dev-env.sre:3306/DevApolloConfigDB?characterEncoding=utf8
    spring.datasource.username = FillInCorrectUser
    spring.datasource.password = FillInCorrectPassword
    eureka.service.url = http://statefulset-apollo-config-server-dev-0.service-apollo-meta-server-dev:8080/eureka/,http://statefulset-apollo-config-server-dev-1.service-apollo-meta-server-dev:8080/eureka/,http://statefulset-apollo-config-server-dev-2.service-apollo-meta-server-dev:8080/eureka/

#创建volumes
      volumes:
        - name: volume-configmap-apollo-admin-server-dev
          configMap:
            name: configmap-apollo-admin-server-dev
            items:
              - key: application-github.properties
                path: application-github.properties
#挂载volumes
          volumeMounts:
            - name: volume-configmap-apollo-admin-server-dev
              mountPath: /apollo-admin-server/config/application-github.properties
              subPath: application-github.properties
```
但是为什么出现read-only报错呢？

kubernetes v1.9.3 之后configmap权限默认是只读[请参考](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#changelog-since-v193)

文件entrypoint.sh中需要创建application-github.properties，实际上在创建容器的时候该文件就已经通过只读方式挂载了，所以会出现read-only错误。

```bash
cat ${admin_server_config}.cp > ${admin_server_config}
```

综上，数据连接信息通过configmap来注入，就不用entrypoint方式了，这也是kubernetes中推荐的配置和容器解耦的方式。

Reply:
如果确实不需要的话，是否可以直接删除相关的行以及文件？

Reply:
是的 已经删除提交了